blueocean-dashboard/src/main/js/components/branchSource/services/BranchPager.js

```diff
@@ -0,0 +1,74 @@
+import { action, computed, observable } from 'mobx';
+import { logging } from '@jenkins-cd/blueocean-core-js';
+import { GenericApi } from '../../rest/GenericApi';
+
+const logger = logging.logger('io.jenkins.blueocean.dashboard.BranchPager');
+const genericApi = new GenericApi();
+
+/**
+ * The pager fetches pages of data from the BlueOcean api. It fetches pages of data, then
+ * inserts them into the [@link BunkerService], and stores the href from the data.
+ *
+ * MobX computes a data field from the href backed by the backend cache. This allows for SSE events
+ * to be propagated to the pager.
+ *
+ * @export
+ * @class Pager
+ */
+export class BranchPager {
```

Comment:
We do not have enough information for the linked to be displayed so we need to get it from somewhere.